pmm/pmm2-server-autobuild.groovy

```diff
@@ -0,0 +1,176 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+]) _
+
+pipeline {
+    environment {
+        DESTINATION = 'laboratory'
+    }
+    agent {
+        label 'large-amazon'
+    }
+    parameters {
+        string(
+            defaultValue: 'PMM-2.0',
+            description: 'Tag/Branch for percona-images repository',
+            name: 'GIT_BRANCH')
+    }
+    options {
+        skipDefaultCheckout()
+        disableConcurrentBuilds()
+    }
+    triggers {
+        upstream upstreamProjects: 'pmm2-submodules-rewind', threshold: hudson.model.Result.SUCCESS
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                installDocker()
+
+                git poll: true, branch: GIT_BRANCH, url: 'http://github.com/Percona-Lab/pmm-submodules'
+                sh '''
+                    git reset --hard
+                    git clean -xdf
+                    git submodule update --init --jobs 10
+
+                    git rev-parse --short HEAD > shortCommit
+                    echo "UPLOAD/${DESTINATION}/${JOB_NAME}/pmm/\$(cat VERSION)/${GIT_BRANCH}/\$(cat shortCommit)/${BUILD_NUMBER}" > uploadPath
+                '''
+                archiveArtifacts 'uploadPath'
+                stash includes: 'uploadPath', name: 'uploadPath'
+                archiveArtifacts 'shortCommit'
+                slackSend channel: '#pmm-ci', color: '#FFFF00', message: "[${JOB_NAME}]: build started - ${BUILD_URL}"
+            }
+        }
+        stage('Build client source') {
+            steps {
+                sh '''
+                    sg docker -c "
+                        export pmm_version=$(cat VERSION)
```

Comment:
Why lowercase? Is it different from `PMM_VERSION`?
And where `VERSION` file is created?

Reply:
I think, It is just to use this as an artifact on different stages

Reply:
pmm_version variable is used to set version of pmm. and VERSION file is in pmm-submodules repo

Reply:
So is it the same as `PMM_VERSION`? Can we use a consistent way of writing it then? Otherwise, it looks like it is a related but different thing.

Reply:
Wait. Nevermind. :) It is `pmm_version` everywhere. I was confusing it with `PMM_RELEASE_VERSION`.